### PR TITLE
NEW add hidden option MAIN_DOC_UPLOAD_NOT_RENAME_BY_DEFAULT

### PR DESCRIPTION
**Pull request by iouston**
# New add MAIN_DOC_UPLOAD_NOT_RENAME_BY_DEFAULT
add a global variable for disable the auto renaming on upload file
corresponds to the request of this post : https://www.dolibarr.fr/forum/12-howto--aide/55094-fichiers-joints-decocher-la-case-sauver-sous#107392

### CODE REVIEW

File: htdocs/core/class/html.formfile.class.php [hunk 1]
```diff
@@ -72,7 +72,7 @@ function __construct($db)
 	 *  @param	string		$options		Add an option column
 	 *  @param	integer		$useajax		Use fileupload ajax (0=never, 1=if enabled, 2=always whatever is option).
      *                                      Deprecated 2 should never be used and if 1 is used, option should no be enabled.
-	 *  @param	string		$savingdocmask	Mask to use to define output filename. For example 'XXXXX-__YYYYMMDD__-__file__'
+	 *  @param	string		$dodocmask	Mask to use to define output filename. For example 'XXXXX-__YYYYMMDD__-__file__'
```
Comment (eldy) [inline htdocs/core/class/html.formfile.class.php]: Why did you change the do here ?

**Comment by iouston** [comment]
Sorry oops, it was certainly a fake manipulation. I'm not very familiar
with these tools


*Julien Marchand*
24, rue de l'Union
54370 Einville au Jard

03.72.59.00.50 - 06.74.83.52.28
julien.marchand@iouston.com
www.iouston.com

**Comment by eldy** [comment]
Do you want me to restore the good var name on your existing PR or do you prefer to repush the PR yourself once fixed ?

**Comment by iouston** [comment]
no, you can restore the good var name
sorry


*Julien Marchand*
24, rue de l'Union
54370 Einville au Jard

03.72.59.00.50 - 06.74.83.52.28
julien.marchand@iouston.com
www.iouston.com
